server/controllers/list.js

```diff
@@ -1233,6 +1233,57 @@ const deleteItem = (req, resp) => {
     .catch(() => resp.sendStatus(400));
 };
 
+const leaveList = (req, resp) => {
+  const { id: listId } = req.params;
+  const { _id: userId } = req.user;
+  const sanitizedListId = sanitize(listId);
+
+  List.findOne({
+    _id: sanitizedListId,
+    viewersIds: userId,
+    type: ListType.LIMITED
+  })
+    .exec()
+    .then(list => {
+      if (!list) {
+        throw new BadRequestException();
+      }
+
+      const { viewersIds, memberIds, ownerIds } = list;
+      const isOwner = checkIfArrayContainsUserId(ownerIds, userId);
+      const isMember = checkIfArrayContainsUserId(memberIds, userId);
+      const isViewer = checkIfArrayContainsUserId(viewersIds, userId);
```

Comment:
maybe it would be better to create methods like: 'isOwner(list, userId)`, `isMember(list, userId)` and use them in `if` statements?

Reply:
We had that before. 
But it was repeatable, every method has same code inside but was named differently.
So about 2 months ago we merged all these 3 methods into one.

Reply:
you can always create methods like this:
```
const isOwner = (list, userId) => checkIfArrayContainesUserId(list, userId)
```